src/EditorFeatures/CSharpTest/Completion/CompletionProviders/TypeImportCompletionProviderTests.cs

```diff
@@ -907,6 +907,270 @@ class Bat
             AssertRelativeOrder(new List<string>() { "SomeType", "SomeTypeWithLongerName" }, completionList.Items);
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(35540, "https://github.com/dotnet/roslyn/issues/35540")]
+        public async Task AttributeTypeInAttributeNameContext()
+        {
+            var file1 = @"
+namespace Foo
+{
+    public class MyAttribute : System.Attribute { }
+    public class MyClass { }
+}";
+
+            var file2 = @"
+namespace Test
+{
+    [$$
+    class Program { }
+}";
+            var markup = CreateMarkupForSingleProject(file2, file1, LanguageNames.CSharp);
+
+            await VerifyTypeImportItemExistsAsync(markup, "My", glyph: (int)Glyph.ClassPublic, inlineDescription: "Foo", expectedDescriptionOrNull: "class Foo.MyAttribute");
+            await VerifyTypeImportItemIsAbsentAsync(markup, "MyAttribute", inlineDescription: "Foo");
+            await VerifyTypeImportItemIsAbsentAsync(markup, "MyClass", inlineDescription: "Foo");
+        }
+
+        [InlineData(SourceCodeKind.Regular)]
+        [InlineData(SourceCodeKind.Script)]
+        [WpfTheory, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(35540, "https://github.com/dotnet/roslyn/issues/35540")]
+        public async Task CommitAttributeTypeInAttributeNameContext(SourceCodeKind kind)
```

Comment:
Do the VerifyExists/VerifyAbsent tests and the Commit tests need to be separated like this? Is the goal to not duplicate the Exists/Absent tests between the `SourceCodeKind`s?

Reply:
It's probably not necessary. Can't remember why I did this, likely because I wrote existence tests first and then commit tests...